subsys/fs/fat_fs.c

```diff
@@ -63,7 +65,20 @@ static int translate_error(int error)
 	return -EIO;
 }
 
-static int fatfs_open(struct fs_file_t *zfp, const char *file_name)
+static uint8_t translate_flags(fs_mode_t flags)
+{
+	uint8_t fat_mode = 0;
+
+	fat_mode |= (flags & FS_O_READ) ? FA_READ : 0;
+	fat_mode |= (flags & FS_O_WRITE) ? FA_WRITE : 0;
+	fat_mode |= (flags & FS_O_CREATE) ? FA_OPEN_ALWAYS : 0;
+	fat_mode |= (flags & FS_O_APPEND) ? FA_SEEKEND : 0;
```

Comment:
This should be named `FAX_APPEND`.  `FA_SEEKEND` is Fat-FS only and means "seek to end of file on open".  That's not something Zephyr supports, so we should use a dedicated name, with a prefix that doesn't conflict with what FatFS uses for its supported modes.

Reply:
Unfortunately "the hack" with using flag from FAT FS would not work, so I have decided to modify `fs_file_t` structure to contain copy of the flags and now all fs_ functions have access to original flags and can use them for different purposes.